vktrace/vktrace_layer/vktrace_lib_trim.cpp

```diff
@@ -1484,6 +1484,190 @@ ObjectInfo &add_Device_object(VkDevice var) {
 }
 
 //=========================================================================
+// The method can be used to query all specific type vulkan objects belong
+// to a device. The query result will be packed into an input vector and
+// return to the caller.
+//
+// \param result_objects, the input vector which is used to store the query
+// result.
+//
+// \param object_map, the input map object which include trim track info of
+// all vulkan objects of the specific type, the method will check every one
+// item in this map to find out which object is created on the device.
+//
+// \param var, the input vulkan device, the method will query all specific
+// type objects belong to this device.
+//
+// \tparam object_class, this is the class type that the method query all
+// objects of this type, it can be QueryPool, Event, Fence and all trim
+// tracked vulkan classes.
+//=========================================================================
+template <class object_class>
+void get_device_objects(std::vector<object_class> &result_objects, std::unordered_map<object_class, ObjectInfo> object_map,
+                        VkDevice var) {
+    for (auto info = object_map.begin(); info != object_map.end(); ++info) {
+        if (info->second.belongsToDevice == var) {
+            result_objects.push_back(reinterpret_cast<object_class>(info->first));
+        }
+    }
+}
+
+//=========================================================================
+// The method can be used to delete all vulkan objects that currently belong
+// to the device. For every trim tracked objects belong to the device, the
+// method delete all these objects one type by one type. it first delete
+// all QueryPool objects, then Event, Fence, Semaphore......CommandPool.
+//
+// For every class type, it first query all objects of this class for the
+// device, then delete all of them by the query result.
+//
+// \param device, this is the device that this method will delete all
+// assossiate objects based on it.
+//=========================================================================
+void delete_objects_for_destroy_device(VkDevice device) {
```

Comment:
Is the types of objects this method deletes complete? How did you create the list?